### PR TITLE
Fix up secondary action button

### PR DESCRIPTION
**Pull request by alex-page**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-ux/issues/520

### WHAT is this pull request doing?

Adding a text color so destructive actions are also reset.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

1. Go to the details page and the `View` button should not have white text.

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [x] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [x] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss [hunk 1]
```diff
@@ -9,6 +9,7 @@ $button-spacing: rem(12px);
   button {
     @include focus-ring($border-width: rem(1px));
     background: transparent !important;
+    color: var(--p-text) !important;
```
Comment (alex-page) [inline src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss:12]: This is kind of weird as it makes the button look not destructive. Should we remove the ability for destructive secondary actions?

Comment (kyledurand) [reply src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss:12]: Can primary be destructive still? If so then I think we should remove primary from the allowed props here

Comment (dleroux) [reply src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss:12]: There are instances where the secondary action is destructive. Pretty sure I've need that.

**Comment by dleroux** [comment]
While I was tophatting your other PR I just happen to use this one for comparison. This one has the opposite problem.

Screen Shot 2021-01-14 at 7 46 27 AM

File: src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss (hunk 1 above)
Comment (avocadomayo) [reply src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss:12]: @alex-page this is what you're referring to, right? The "View" action which is secondary doesn't have a differentiating colour?

Screen Shot 2021-01-13 at 4 43 13 PM

And seconding @dleroux - I've needed a destructive secondary action in `ActionMenu` on several occasions

Comment (alex-page) [reply src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss:12]: Yeah exactly the `View` button looks like the other secondary actions. We might need to get some additional UX opinions but I wonder if the balance a destructive button in the page would bring too much attention, if that is the case should it look the same or is there a way to make it look destructive without it being so distracting.

**Comment by alex-page** [comment]
> While I was tophatting your other PR I just happen to use this one for comparison. This one has the opposite problem.
> 
> Screen Shot 2021-01-14 at 7 46 27 AM

This is fixed in master, I will merge it into this branch now.

**Comment by tomoyukikashiro** [comment]
@dleroux Can this PR be merged ? I need destructive secondaryAction button!

**Pull request closed without merging**